src/main/java/org/wildfly/security/http/impl/ClientCertAuthenticationMechanism.java

```diff
@@ -104,18 +107,27 @@ private boolean attemptAuthentication(HttpServerRequest request, Function<Securi
         X509Certificate[] x509Certificates = X500.asX509CertificateArray(peerCertificates);
         X509PeerCertificateChainEvidence evidence = new X509PeerCertificateChainEvidence(x509Certificates);
 
-        httpClientCert.tracef("Authenticating using following certificates: [%s]", x509Certificates);
+        if (httpClientCert.isTraceEnabled()) {
+            httpClientCert.tracef("Authenticating using following certificates: [%s]", Arrays.toString(x509Certificates));
+        }
 
-        EvidenceVerifyCallback callback = new EvidenceVerifyCallback(evidence);
-        try {
-            MechanismUtil.handleCallbacks(httpClientCert, callbackHandler, callback);
-        } catch (AuthenticationMechanismException e) {
-            throw e.toHttpAuthenticationException();
-        } catch (UnsupportedCallbackException e) {
-            throw httpClientCert.mechCallbackHandlerFailedForUnknownReason(e).toHttpAuthenticationException();
+        boolean verified;
+        if (! skipVerification) {
+            EvidenceVerifyCallback callback = new EvidenceVerifyCallback(evidence);
+            try {
+                MechanismUtil.handleCallbacks(httpClientCert, callbackHandler, callback);
+            } catch (AuthenticationMechanismException e) {
+                throw e.toHttpAuthenticationException();
+            } catch (UnsupportedCallbackException e) {
+                throw httpClientCert.mechCallbackHandlerFailedForUnknownReason(e).toHttpAuthenticationException();
+            }
+            verified = callback.isVerified();
+            httpClientCert.tracef("X509PeerCertificateChainEvidence was verified by EvidenceVerifyCallback handler: %b", verified);
+        } else {
+            verified = true;
```

Comment:
This is just one small objection against readability. Here  verified = true is misleading as it is in fact not verified.